llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -9800,3 +9817,45 @@ SDValue SystemZTargetLowering::lowerVECREDUCE_ADD(SDValue Op,
       ISD::EXTRACT_VECTOR_ELT, DL, VT, DAG.getBitcast(OpVT, Op),
       DAG.getConstant(OpVT.getVectorNumElements() - 1, DL, MVT::i32));
 }
+
+// Only consider a function fully internal as long as it has local linkage
+// and is not used in any other way than acting as the called function at
+// call sites.  TODO: Remove this when/if all internal functions adhere to
+// the ABI.
```

Comment:
We're generally free to change the ABI of internal functions, including in ways that do not match the system ABI. So I don't think this TODO can ever really be resolved.

Reply:
OK, removing that for now at least (not sure if this is actually the case on SystemZ that we would ever make special internal ABIs when lowering calls/returns).